### PR TITLE
[Feature] Add two traditional filters to smooth poses

### PR DESCRIPTION
**Pull request by ailingzengzzz**
## Motivation

Smooth poses.

## Modification

1. Add `Smoother` module for pose smoothing, which applies a temporal filter on pose sequence.
2. Support 3 filters: Savitzky-Golay 2d filter, Gaussian 1d filter, and One-Euro filter (derived from the existing implementation in MMpose)
3. Update inference APIs, demos, and Webcam APIs.
4. Add unit tests of smoother and filters
5. Fix the bug reported in #1176 

### Init smoother

```python
from mmpose.core import Smoother

smoother = Smoother(filter_cfg='configs/_base_/filters/one_euro.py', keypoint_dim=2)
```

### Online smoothing (frame-by-frame)

```python
for frame in video:
    # pose estimation
    results = inference_top_down_pose_model(model, frame, ...)
    # pose tracking
    results, next_id = get_track_id(results, last_results, next_id, ...)
    # smoothing
    results_smoothed = smoother.smooth(results)
```

### Offline smoothing

```python
results_sequence = []

for frame in video:
    # pose estimation
    results = inference_top_down_pose_model(model, frame, ...)
    # pose tracking
    results, next_id = get_track_id(results, last_results, next_id, ...)

    results_sequence.append(results)

results_sequence_smoothed = smoother.smooth(results_sequence)
```

## BC-Breaking

The arguments `use_one_euro` and `fps` in function `get_track_id()` will be deprecated in the future. Deprecation warnings have been added in this PR.

## Checklist

**Before PR**:

- [ ] I have read and followed the workflow indicated in the [CONTRIBUTING.md](https://github.com/open-mmlab/mmpose/blob/master/.github/CONTRIBUTING.md) to create this PR.
- [ ] Pre-commit or linting tools indicated in [CONTRIBUTING.md](https://github.com/open-mmlab/mmpose/blob/master/.github/CONTRIBUTING.md) are used to fix the potential lint issues.
- [ ] Bug fixes are covered by unit tests, the case that causes the bug should be added in the unit tests.
- [ ] New functionalities are covered by complete unit tests. If not, please add more unit tests to ensure correctness.
- [ ] The documentation has been modified accordingly, including docstring or example tutorials.

**After PR**:

- [ ] CLA has been signed and all committers have signed the CLA in this PR.

### CODE REVIEW

File: .gitignore [hunk 1]
```diff
@@ -125,8 +125,14 @@ docs_zh-CN/topics/
 docs_zh-CN/papers/*.md
 docs_zh-CN/datasets.md
 docs_zh-CN/modelzoo.md
-
+mmcv/
```
Comment (jin-s13) [inline .gitignore:128]: The modification of this file may be not necessary.

File: mmpose/core/post_processing/__init__.py [hunk 2]
```diff
@@ -1,6 +1,10 @@
 # Copyright (c) OpenMMLab. All rights reserved.
 from .nms import oks_iou, oks_nms, soft_oks_nms
-from .one_euro_filter import OneEuroFilter
+#from .one_euro_filter import OneEuroFilter
```
Comment (jin-s13) [inline mmpose/core/post_processing/__init__.py:3]: Delete unused comments.

File: .gitignore (hunk 1 above)
Comment (ailingzengzzz) [reply .gitignore:128]: Sure. You can simply delete them.

File: mmpose/core/post_processing/__init__.py (hunk 2 above)
Comment (ailingzengzzz) [reply mmpose/core/post_processing/__init__.py:3]: This is a previous version of One-Euro filter. I'm not sure whether to leave it or not.

**Comment by jin-s13** [comment]
Pls follow https://github.com/open-mmlab/mmpose/blob/master/.github/CONTRIBUTING.md to setup pre-commit.

File: mmpose/core/post_processing/__init__.py [hunk 3]
```diff
@@ -1,6 +1,9 @@
 # Copyright (c) OpenMMLab. All rights reserved.
 from .nms import oks_iou, oks_nms, soft_oks_nms
-from .one_euro_filter import OneEuroFilter
+from .builder import build_filter
+from .gaus1d_filter import Gaus1dFilter
```
Comment (jin-s13) [inline mmpose/core/post_processing/__init__.py:4]: How about gauss1d_filter

File: mmpose/core/post_processing/gaus1d_filter.py [hunk 4]
```diff
@@ -0,0 +1,59 @@
+import warnings
+
+import numpy as np
+import scipy.signal as signal
+import torch
+from scipy.ndimage.filters import gaussian_filter1d
```
Comment (jin-s13) [inline mmpose/core/post_processing/gaus1d_filter.py:6]: It would be much better if this func can be implemented in pytorch.

Comment (jin-s13) [reply mmpose/core/post_processing/gaus1d_filter.py:6]: Just saying. For now, that's fine.

File: mmpose/core/post_processing/__init__.py (hunk 3 above)
Comment (ailingzengzzz) [reply mmpose/core/post_processing/__init__.py:4]: Okay, I will modify it.

**Review by ly015 — APPROVED** [review]

File: mmpose/core/post_processing/temporal_filters/one_euro_filter.py [hunk 5]
```diff
@@ -0,0 +1,110 @@
...
+    r = 2 * math.pi * cutoff * t_e
+    return r / (r + 1)
+
+
+def exponential_smoothing(a, x, x_prev):
+    return a * x + (1 - a) * x_prev
+
+
+class OneEuro:
+
```
Comment (jin-s13) [inline mmpose/core/post_processing/temporal_filters/one_euro_filter.py:23]: Do we need any description here?

**Review by jin-s13 — APPROVED** [review]

File: mmpose/core/post_processing/temporal_filters/one_euro_filter.py (hunk 5 above)
Comment (ly015) [reply mmpose/core/post_processing/temporal_filters/one_euro_filter.py:23]: This is part of the newly reimplemented One-Euro filter. The one that will be deprecated is in `mmpose/core/post_processing/one_euro_filter.py`.

**Comment by OpenMMLab-Assistant-004** [comment]
Hi @ailingzengzzz ！First of all, we want to express our gratitude for your significant PR in the mmpose project. Your contribution is highly appreciated, and we are grateful for your efforts in helping improve this open-source project during your personal time. We believe that many developers will benefit from your PR.

We would also like to invite you to join our Special Interest Group (SIG) private channel on Discord, where you can share your experiences, ideas, and build connections with like-minded peers. To join the SIG channel, simply message moderator— OpenMMLab on Discord or briefly share your open-source contributions in the #introductions channel and we will assist you. Look forward to seeing you there! Join us ：https://discord.gg/raweFPmdzG

If you have WeChat，welcome to join our community on WeChat. You can add our assistant ：openmmlabwx. Please add "mmsig + Github ID" as a remark when adding friends：）
Thank you again for your contribution❤

**Pull request merged**